### ISSUE TITLE
Automate guarantee that mk8sv is satisified

### ISSUE THREAD

**Issue by aviramha** [kube-rs/kube#1611]
Followup from:
> I think this breaks the Kubernetes version guarantee - requires openapi to upgraded to 
 1.31 to compile kube-rs as a dependency. @clux

_Originally posted by @aviramha in https://github.com/kube-rs/kube/issues/1605#issuecomment-2421919001_
            

An innocent change broke that guarantee and could've slipped to a release.
Opening this issue to discuss how we can guarantee that this doesn't happen in the future.
My suggestion is to change the dev dependency of openapi to use mk8sv (perhaps run tests both on latest and mk8sv?)

Happy to hear other suggestions

**Comment by clux** [comment]
It's funny because we [do run the tests on both versions](https://github.com/kube-rs/kube/blob/1addf43fd6fe76795b5a26eed715b8815b8826a5/.github/workflows/ci.yml#L136-L236), but there's clearly a gap here since this got through.

Thanks for raising.

**Comment by aviramha** [comment]
Yeah it's runtime vs compilation - the new version will run on older clusters, just won't compile with "old" openapi

**Comment by clux** [comment]
Yeah. Tbh, this is a reason why parts of me wants to drop k8s-openapi's version choice and turn it into fixed `latest` because the runtime is way more important than the schemas. If we pin `latest` we can:

- drop all the feature madness, e2e tests actualling un-defaulting the feature, and adding in earlier
- re-export k8s-openapi exports under a namespace (as in [here](https://github.com/kube-rs/kube/pull/1602/files#diff-fb122394c4efbe65ed23a4d0b8c53862a08a9c0d93f4013709ed436337c48479))
- avoid people having to update k8s-openapi and kube together (we pin, you get updates)

It's also the versioning strategy most other kubernetes tools on the go side uses (update client-go? cool, dont' stray too far, here's a compatibility matrix, but you get updated structs anyway).

The alternative here is doing more like what we do in the `e2e` style tests (which do actually run both feature versions).

**Comment by aviramha** [comment]
Okay, sounds like a good plan to me. It's less backward compatible but good trade offs in terms of usability/maintenance and if it's the standard then no reason to stray from that.
